components/slider-carousel/src/ReuseSlider.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { useState, useEffect, useRef, CSSProperties } from "react";
+import SliderProgress from "./Progress";
+
+interface statusInterface {
+  statusArray: React.ReactNode[];
+  slideInterval?: number;
+  progressStyle?: string | CSSProperties;
+  progressContainerStyles?: string | CSSProperties;
+  SliderContainerStyles?: string | CSSProperties;
+  sliderWrapperClasses?: string;
+  footer?: React.ReactNode;
+  loop?: boolean;
+  nextButton?: React.ReactNode;
+  prevButton?: React.ReactNode;
+  enableButton?: boolean;
+  progressPosition?: string;
+}
+const nextBtn = () => {
+  return (
+    <div className="absolute top-1/2 bottom-1/2 right-5 flex -rotate-270 p-4 justify-center items-center rounded-full bg-black opacity-75 h-10 w-10 z-10">
+      <span>&#8811;</span>
+    </div>
+  );
+};
+
+const prevBtn = () => {
+  return (
+    <div className="absolute top-1/2 bottom-1/2 left-5 flex rotate-270 p-4 justify-center items-center rounded-full bg-black opacity-75 h-10 w-10 z-10">
+      <span>&#8810;</span>
+    </div>
+  );
+};
+
+const ReuseSlider = ({
+  slideInterval = 2000,
+  statusArray,
+  progressStyle,
+  progressContainerStyles,
+  SliderContainerStyles,
+  sliderWrapperClasses,
+  progressPosition,
+  footer,
+  loop,
+  enableButton = false,
+  nextButton = nextBtn(),
+  prevButton = prevBtn(),
+}: statusInterface) => {
+  const [currentStatusIndex, setCurrentStatusIndex] = useState(0);
+  const [isScrollPaused, setScrollPaused] = useState(false);
+  const [remainingPauseTime, setRemainingPauseTime] = useState<any>(null);
+  const [preIndex, setPreIndex] = useState<number>();
+  const [lastTime, setLastTime] = useState<number>(0);
+  const [slideChange, setSlideChange] = useState<boolean>(false);
+
+  const intervalRef = useRef<any>(null);
+  const currentStatus = statusArray[currentStatusIndex];
+
+  let remainingTime =
+    preIndex == currentStatusIndex && lastTime > 0
+      ? slideInterval - lastTime
+      : slideInterval;
+
+  const startAutoScroll = () => {
+    clearInterval(intervalRef.current);
+    if (!isScrollPaused && loop) {
+      setCurrentStatusIndex((preindex) => {
+        const nextIndex = preindex == statusArray.length - 1 ? 0 : preindex + 1;
+        return nextIndex;
+      });
+    } else if (!isScrollPaused && !loop) {
+      if (currentStatusIndex < statusArray.length - 1) {
+        setCurrentStatusIndex((preindex) => {
+          const nextIndex = preindex + 1;
+          return nextIndex;
+        });
+      } else if (currentStatusIndex == statusArray.length - 1) {
+        setCurrentStatusIndex(currentStatusIndex);
+      }
+      if (currentStatusIndex != statusArray.length - 1) {
+      }
+    }
+  };
+
+  const handlePrevStatus = () => {
```

Comment:
A lot of code in `handlePrevStatus` and `handleNextStatus` is repeated. Can it be extracted into a separate function?